From 40b2f203bb0691f45d024c6bf59433cdb5712160 Mon Sep 17 00:00:00 2001
From: Giovanni Campagna <gcampagna@src.gnome.org>
Date: Sun, 25 Nov 2012 23:51:44 +0100
Subject: [PATCH 3/3] MessageTray: remove all tweens when tweening for state
 parameters

MessageTray._tween sets the state variable to the in-progress value,
so it must be sure that at the end of the animation the value will
be the corresponding final and nothing else will happen in between.

https://bugzilla.gnome.org/show_bug.cgi?id=683986
---
 js/ui/messageTray.js |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/js/ui/messageTray.js b/js/ui/messageTray.js
index 7079c19..e36f499 100644
--- a/js/ui/messageTray.js
+++ b/js/ui/messageTray.js
@@ -2051,6 +2051,8 @@ const MessageTray = new Lang.Class({
         params.onCompleteScope = this;
         params.onCompleteParams = [statevar, value, onComplete, onCompleteScope, onCompleteParams];
 
+        // Remove other tweens that could mess with the state machine
+        Tweener.removeTweens(actor);
         Tweener.addTween(actor, params);
 
         let valuing = (value == State.SHOWN) ? State.SHOWING : State.HIDING;
-- 
1.7.10.4

